app/src/sharedTest/java/org/oppia/app/parser/StringToRatioParserTest.kt

```diff
@@ -0,0 +1,104 @@
+package org.oppia.app.parser
+
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.app.model.RatioExpression
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+
+/** Tests for [StringToRatioParser]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(manifest = Config.NONE)
+class StringToRatioParserTest {
+
+  private lateinit var stringToRatioParser: StringToRatioParser
+  private lateinit var context: Context
+
+  @Before
+  fun setUp() {
+    context = ApplicationProvider.getApplicationContext()
+    stringToRatioParser = StringToRatioParser()
+  }
+
+  @Test
+  fun testParser_realtimeError_returnInvalidCharsError() {
+    val error =
+      stringToRatioParser.getRealTimeAnswerError("abc").getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo(
+      "Please write a ratio that consists of digits separated by colons (e.g. 1:2 or 1:2:3)."
+    )
+  }
+
+  @Test
+  fun testParser_realtimeError_returnInvalidFormatError() {
+    val error =
+      stringToRatioParser.getRealTimeAnswerError("1:2:3:").getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo("Please enter a valid ratio (e.g. 1:2 or 1:2:3).")
+  }
+
+  @Test
+  fun testParser_realtimeError_returnInvalidColonsError() {
+    val error = stringToRatioParser.getRealTimeAnswerError("1::2")
+      .getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo("Your answer has two colons (:) next to each other.")
+  }
+
+  @Test
+  fun testParser_realtimeError_returnValid() {
+    val error = stringToRatioParser.getRealTimeAnswerError("1:2:3").name
```

Comment:
Shouldn’t the error be null if there isn’t one?

Reply:
updated